element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java

```diff
@@ -339,16 +340,23 @@ protected void work() throws IOException {
 			DatagramSocket currentSocket = getSocket();
 			if (currentSocket != null) {
 				currentSocket.receive(datagram);
-				LOGGER.debug("UDPConnector ({}) received {} bytes from {}:{}", effectiveAddr, datagram.getLength(),
-						datagram.getAddress(), datagram.getPort());
-				byte[] bytes = Arrays.copyOfRange(datagram.getData(), datagram.getOffset(), datagram.getLength());
-				RawData msg = RawData.inbound(bytes,
-						new UdpEndpointContext(new InetSocketAddress(datagram.getAddress(), datagram.getPort())),
-						false);
-				receiver.receiveData(msg);
+				if (datagram.getLength() >= size) {
+					// too large datagram for our buffer! data could have been
+					// truncated, so we discard it.
+					LOGGER.debug(
+							"UDPConnector ({}) received truncated UDP datagram from {}:{}. Maximum size allowed {}. Discarding ...",
+							effectiveAddr, datagram.getAddress(), datagram.getPort(), size - 1);
```

Comment:
In some cases a RST or 4.xx response may be preferred to stop retransmissions and inform the sender, but though I don't think it will be triggered too often, it's OK.

Reply:
Wait ! for DTLS :thinking:, if record is truncated we will not be able to decrypt it, right ?

So, what I proposed is maybe not doable in DTLS ? so I don't know if it is worthy to do it just for unsecured CoAP.

Reply:
> Wait ! for DTLS thinking, if record is truncated we will not be able to decrypt it, right ?

Yeap.

> And so Upper layer could try to decode&reply to it.

I would only do that, if someone complains ...
I guess, in the most cases it will not be the californium peer, which has trouble to decode too large messages :-).